hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -790,23 +797,28 @@ public Cell getCell() {
         // we can handle the 'no tags' case.
         if (currTagsLen > 0) {
           ret = new SizeCachedKeyValue(blockBuffer.array(),
-              blockBuffer.arrayOffset() + blockBuffer.position(), cellBufSize, seqId);
+              blockBuffer.arrayOffset() + blockBuffer.position(), cellBufSize, seqId, currKeyLen,
+              rowLen);
         } else {
           ret = new SizeCachedNoTagsKeyValue(blockBuffer.array(),
-              blockBuffer.arrayOffset() + blockBuffer.position(), cellBufSize, seqId);
+              blockBuffer.arrayOffset() + blockBuffer.position(), cellBufSize, seqId, currKeyLen,
+              rowLen);
         }
       } else {
         ByteBuffer buf = blockBuffer.asSubByteBuffer(cellBufSize);
         if (buf.isDirect()) {
-          ret = currTagsLen > 0 ? new ByteBufferKeyValue(buf, buf.position(), cellBufSize, seqId)
-              : new NoTagsByteBufferKeyValue(buf, buf.position(), cellBufSize, seqId);
+          ret = currTagsLen > 0
+              ? new SizeCachedByteBufferKeyValue(buf, buf.position(), cellBufSize, seqId,
+                  currKeyLen, rowLen)
+              : new SizeCachedNoTagsByteBufferKeyValue(buf, buf.position(), cellBufSize, seqId,
+                  currKeyLen, rowLen);
         } else {
           if (currTagsLen > 0) {
             ret = new SizeCachedKeyValue(buf.array(), buf.arrayOffset() + buf.position(),
-                cellBufSize, seqId);
+                cellBufSize, seqId, currKeyLen, rowLen);
```

Comment:
Ya here also as currKeyLen is already calculated we can pass.. But let rowLen be decoded within SizeCachedKeyValue only.  We can avoid all these changes

Reply:
Here too the rowLen is already there. Any specific reason you feel this need not be decoded.

Reply:
Any advantage we get because of doing here?  Because in one place, while constructing this KV objects, we decode the value and set as a state var and pass in constructor.  For keyLen, ya we already decoded there (before this patch) we pass that for reuse.  But for rowLen its not that way na..  So why to change the critical part of HFile reader adding new byte[] state? Better avoid.

Reply:
If you see my previous commit - I got the rowLen also in blockSeek and the readKeyValue() method to pass it to the SizeCachedKV variants. That was mainly just to ensure that we get rowLen while doing the KV parsing itself rather than KV creation. Now in the last commit since I was already doing that rowLen related change I changed the BBKV also so that we don't need to parse it there.

Reply:
>>So why to change the critical part of HFile reader adding new byte[] state
Sorry. When you mean byte[] state - are you saying we are adding a new byte[] as state variable?

Reply:
Sorry not rowKey... Ya new rowLen state. My bad.  What is the real adv of this rowLen decoding happening at the Reader class?(and set to a state var)